CerebNet/run_prediction.py

```diff
@@ -17,18 +17,45 @@
 # IMPORTS
 import sys
 import argparse
+from typing import Dict
 
 from FastSurferCNN.utils import logging, parser_defaults
 
 from CerebNet.utils.load_config import get_config
 from CerebNet.inference import Inference
-from FastSurferCNN.utils.checkpoint import get_checkpoints
+from FastSurferCNN.utils.checkpoint import get_checkpoints, get_plane_dict, PLANE_DEFAULTS
 from FastSurferCNN.utils.common import assert_no_root, SubjectList
 
 logger = logging.get_logger(__name__)
 DEFAULT_CEREBELLUM_STATSFILE = "stats/cerebellum.CerebNet.stats"
 
 
+def merge_dict_in_args(args: argparse.Namespace, dictionary: Dict, defaults: Dict) -> argparse.Namespace:
+    wrong_files = []
+
+    for plane in ["axial", "coronal", "sagittal"]:
+        # find the first vowel in the key
+        index = min(i for i in (short_plane.find(v) for v in "aeiou") if i >= 0)
+        short_plane = short_plane[: index + 2]
+        if getattr(args, f"ckpt_{short_plane}") != dictionary["CEREBNET"]["CKPT"][plane.upper()]:
+            if getattr(args, f"ckpt_{short_plane}") == defaults["CEREBNET"]["CKPT"][plane.upper()]:
+                setattr(args, f"ckpt_{short_plane}", dictionary["CEREBNET"]["CKPT"][plane.upper()])
+            else:
+                wrong_files.append(f"ckpt {plane}")
+    
+    if getattr(args, f"cfg_file") != dictionary["CEREBNET"]["CFG"]["UNIVERSAL"]:
+            if getattr(args, f"cfg_file") == defaults["CEREBNET"]["CFG"]["UNIVERSAL"]:
+                setattr(args, f"cfg_file", dictionary["CEREBNET"]["CFG"]["UNIVERSAL"])
+            else:
+                wrong_files.append(f"cfg_file")
+
+
+    if len(wrong_files) > 0:
+        raise ValueError("The following files were given twice, once in the command line and once in the yaml file, but do not match:"
+            + ", ".join(wrong_files))
+
+    return args
+
 def setup_options():
```

Comment:
Can you elaborate why the loading would not otherwise work properly unless we use a custom parser class instead of the usual one with added arguments? Also this sounds like a different feature that goes beyond this change of the default ckpt value. I could work on it, but I would not include it in this PR

Reply:
basically, you want to modify the behavior of how arguments are parsed and replaced, you need that do happen somewhere. This can be done by Actions (from argparse... which can be custom Actions), but here, you want it to be done after all parameters are parsed. So after all parameters are parsed, you want to replace some values if they are still the default -- including a check if the service can be reached and using the backup, if they main server is unreachable.